test_pull_requests

```diff
@@ -1106,6 +1139,18 @@ popd
 
     end
 
+    # The GitHub mergeable API is flaky, so we use an external file ``database'' to record
+    # pull request mergeability responses with a thresholded saturating counter so that pull
+    # request comment/label actions are only taken after a string of consequtive API responses.
+    # The current thresholds are at 0 and 10, with an initial state of 0, so i.e. the default
+    # state is to consider the pull request mergeable, and this state will not change until
+    # a net of ten unmergeable responses come in from the GitHub API. When set_mergeable is
```

Comment:
It's not that the state won't change.  One not mergeable and we immediately won't try to test it.  We just wait to add the comment/label until we are sure.